compliance/collection/nodescanv2/nodescan.go

```diff
@@ -0,0 +1,25 @@
+package nodescanv2
+
+import (
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/logging"
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+// NodeScanner defines an interface for V2 NodeScanning
+type NodeScanner interface {
+	Scan(nodeName string) (*storage.NodeScanV2, error)
```

Comment:
I think the node name should not be passed to the implementation that scans the node. That's the detail that should be filled in `storage.NodeScanV2` by the calling code in Compliance.

Moreover, it seems to me that `NodeScanner` should return components and notes and not care about the rest.

Reply:
The decision was influenced by 2 factors:
1. Scanners `Analyze` function that we will call in the real implementation requires the node name at call time
2. compliance main.go already defines and uses code ([here](https://github.com/stackrox/stackrox/blob/master/compliance/collection/main.go#L36)) to determine the Node name
3. 
I aimed to minimize code changes, but I could as well refactor the existing `getNode` function into the `compliance.utils` package and use it directly inside `NodeScanner.Scan()` instead of passing it from the calling main compliance loop. WDYT?

Reply:
I did not check that node name is a requirement of the real `Analyze()`. Thanks for pointing it out.

I think, it does not need to be a requirement of `Analyze` because the node name is used only as an additional logged field. I suppose eventually this field may be handy during logs analysis for filtering logs collected from all nodes.
Also, it is clear that the node name is used instead of the image layer name in logging because image layers don't exist for a node.

Overall, I don't mind to keep the node name there for now. Building something different to pass this info as some "generic logging field" parameter would result in much larger code with tiny or no benefit.
Let's keep it as-is for now and see where we end up when the real implementation will be used.